### PR TITLE
ref: Changes required to tag Sentry 9.1

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by dcramer** [comment]
I can break these out (integrations is and will land separately), but I'd rather just land this PR as a whole with just rebase+merge.

**Comment by dcramer** [comment]
I tested a 9.0 install with data up to a 9.1 - looks green!

File: tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap [hunk 1]
```diff
@@ -35,7 +35,20 @@ exports[`InstallWizard renders 1`] = `
             errorMessage="Unable to save your changes. Please ensure all fields are valid and try again."
             footerClass="form-actions align-right"
             hideErrors={false}
-            initialData={Object {}}
+            initialData={
+              Object {
+                "auth.allow-registration": undefined,
```
Comment (billyvg) [inline tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap]: Is this right?

Comment (dcramer) [reply tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap]: Let me run one more QA pass to make sure all the values are being respected as we think they should be.

Comment (dcramer) [reply tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap]: everything appears to work as expected when testing by hand so the best I can say is "yes"

Comment (dcramer) [reply tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap]: actually i lied - something on /manage/ settings is causing the form to re-render so need to debug whats not saving correctly

Comment (dcramer) [reply tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap]: there's an issue with the defaults not saving in a way we want, so just hacked around the specific option which needs this behavior

Comment (dcramer) [reply tests/js/spec/views/__snapshots__/installWizard.spec.jsx.snap]: not sure why this snapshot is still the same tbqh... id have expected no changes other than something to beacon.anonymous

**Comment by dcramer** [comment]
good to go if people wanna review - few changes which will need merged into master

**Review by markstory — APPROVED** [review]
Looks good to me.

**Comment by dcramer** [comment]
Added changelog and giant "sorry for the future" notes

**Comment by dcramer** [comment]
@mattrobenolt how do you wanna go about getting this out? I'm going to merge the changes into master, and then we can either split from there, or we can just tag from this branch.

**Pull request merged**